### DISCUSSION TITLE
too many files in resources directory

### DISCUSSION THREAD

**Discussion by kranurag78** [mivinci/hugo-theme-minima#51] [Q&A] [answered]
Hey there, 

![image](https://user-images.githubusercontent.com/84301342/225330104-dbeff729-29fa-4bb7-8dc6-87b6679230a2.png)

I wanted to bring this to your attention. When I started using or tweaking something around this theme then the files under `resources` directory kept growing and I was wondering why this is growing? Every file seems similar `main.scss_<random-hash>.content` and same for `.json` as well. 
I understand we keep `resource` directory in `.gitignore` so it doesn't matter anyways but if we can stop this then that will be super awesome. 
I also use this [theme](https://github.com/hugo-sid/hugo-blog-awesome) but there I don't have faced this problem. I don't have seen this problem with any other theme in the past as well. This is not a comparision (_I'm loving this theme_) but I just wanted to understand the reason behind this and can we do something to mitigate this. If yes, I would love to give it a try. 

Thanks

**Comment by mivinci** [comment]
It is a naive and probably stupid solution to force clear out the browser's cache for websites using this theme when static resources like CSS files are updated.

In file [layouts/partials/head.html](https://github.com/mivinci/hugo-theme-minima/blob/main/layouts/partials/head.html), a timestamp is used as part of the bundled file name, so browsers will always have the newest CSS and JavaScript code immediately every time we have a new release.

```html
{{ $hash := now.Unix }}
{{ $options := (dict "targetPath" (printf `minima.%d.css` $hash) "outputStyle" "compressed" "enableSourceMap" true) }}
{{ $style := resources.Get "css/main.scss" | resources.ToCSS $options }}
<link rel="stylesheet" href="{{ $style.RelPermalink }}">
{{ $options = (dict "targetPath" (printf `minima.%d.js` $hash) "minify" true) }}
{{ $js := resources.Get "js/main.js" | js.Build $options | resources.ExecuteAsTemplate (printf `minima.%d.js` $hash) . }}
<script defer type="text/javascript" src="{{ $js.RelPermalink }}"></script>
```

This is something I need when I first made this theme, and now can be removed from Minima I guess.

**Comment by kranurag78** [comment]
Thanks @mivinci,
I've tried removing the code part that you shared above but there's something happening which I'm not able to figure out. Please see the attached screenshot. 
<details>
<summary> Screenshot </summary>

![image](https://user-images.githubusercontent.com/84301342/225358319-3b955b14-b552-4940-9e81-7fc1a898f05d.png)

</details>

The main reason for asking this because with the lot of files, it taking up space. When I'm running a big blog or a blog with multiple pages then the space will be quite high. 
Here's the data for a blog with 6 pages only on which I'm developing now. 
```bash
$ du -d1 -h
8.0K	./archetypes
36K	./content
736K	./resources
4.0K	./static
4.0K	./data
852K	./themes
4.0K	./public
668K	./.git
4.0K	./layouts
2.3M	.
```
You can notice that `resources` directory is taking up the most space. 
I will be really thankful to you if you can handle this from your side when you've time available for this? I'm still looking into this and trying to understand why the rendering is not happening properly. 

Thanks a ton for the help.

**Comment by mivinci** [comment]
Yea I can understand that it really is annoying since it takes up too much space in there. I'm pushing a commit to fix that right now.

**Comment by mivinci** [comment, marked as answer]
Fixed with commit a0a89797edb1a72851aad06a56fbd997c158ff80

**Comment by kranurag78** [comment]
Hey @mivinci sorry for disturbing you again but I think the issue remains the same. Please look at the attached screenshots. 

<details>
<summary> screenshots </summary>

![image](https://user-images.githubusercontent.com/84301342/225400110-74f19ed5-0646-4008-ab89-d17102cedc0c.png)

---

![image](https://user-images.githubusercontent.com/84301342/225400306-166f7666-e436-49ac-a185-cf098757cbff.png)

</details>

The rendering however is working fine.

**Reply by mivinci** [reply]
hmmm are those files generated before the fix? cuz I only got one file like that after the fix.

**Reply by kranurag78** [reply]
Sorry for the late response. I slept yesterday and in the morning rushed to college. Yes, this was my issue. I forgot to delete everything under resources directory. 

I also wanted to get your opinion on this one. Do you think that code blocks are relatively small as compared to other texts? How about if we bump the font of the code blocks a just a little bit. 
![image](https://user-images.githubusercontent.com/84301342/225601217-83d7175d-8cff-4052-be65-966ed14f367b.png)

**Reply by mivinci** [reply]
It's okay. I'm also busy with college things :)

Actually, the font size of code blocks was a little bit bigger before, but someday I thought these blocks shouldn't be as obvious as the regular text around them, so I made it smaller intentionally.

But it's weird that the one you posted above looks even smaller than those in the example site, and has no paddings at all 🤔 It shouldn't be like that, can you show me your markdown file?

It looks like some CSS attributes have been broken or rewritten cuz your background color of code blocks looks different also.

**Comment by kranurag78** [comment]
Sure, Here's the markdown file the above document. In case this helps, here's the [link](https://kranurag.dev) to the website. 

```text
### After spinning cloud9 here're the things I do

1. Install fish
```bash
sudo yum install fish -y
```

2. Installing Arkade 
```bash
curl -sLS https://get.arkade.dev | sh 
```

3. Get ready with K8s environment 
```bash
ark g minikube kubectl kubens
export PATH=$PATH:$HOME/.akrade/bin 
source ~/.bashrc
```
4. Create a global mark for fish config 
In my local system, I've setup `F` as my mark for fish config which is available at `~/.config/fish/config.fish` and in the ephemeral environment I open up the file and use `mF` to create the marks. 
```bash
$ vim ~/.config/fish/config.fish
$ mA
```


### Usecases 
1. Compiling Rust Programs 
I am using a lot of rust based CLI tools these days and often if they release a new feature then I want to try those things by compiling the program and often this is little difficult on my personal system. So, I spin up a cloud9 instance and then compile the program there and then download the artifact from the cloud9 dashboard itself. 
In the past, I've tried 

I've not configured anything from my side.

**Reply by mivinci** [reply]
Did you modify configs in [here](https://github.com/mivinci/hugo-theme-minima/blob/a0a89797edb1a72851aad06a56fbd997c158ff80/exampleSite/config.yaml#L120)? Cuz the HTML of code blocks in your site look different than that in the example site.

I made this mandatory cuz I want code highlighting to be customizable by CSS variables like [here](https://github.com/mivinci/hugo-theme-minima/blob/a0a89797edb1a72851aad06a56fbd997c158ff80/assets/css/theme.scss#L7).

**Reply by kranurag78** [reply]
Yes, I did some tweaks even after reading the warning because the bash codeblocks here https://kranurag.dev/snippets/git/ was not looking nice. 
I did `hugo gen chromastyles --styles monokai > highlight.css` Let me revert that quickly.

**Reply by mivinci** [reply]
hmmm I didn't expect that anyone would use `hugo gen chromastyles` when I made this theme 😁 Will check out if it is a better solution than mine when I have time. 😬

**Reply by kranurag78** [reply]
I've updated reverted the changes and now it's looking something like this. This one looks pretty small. 
![image](https://user-images.githubusercontent.com/84301342/225615691-69e4fd7e-858c-4a06-ad95-e53f0fcde3fd.png)

**Reply by mivinci** [reply]
Can you show me a link to this page again? Cuz the font size of the example site looks just fine to me. I will see if there's any difference later.

**Reply by kranurag78** [reply]
> Can you show me a link to this page again? Cuz the font size of the example site looks just fine to me. I will see if there's any difference later.

The link can be accessed [here](https://kranurag.dev/snippets/git/)

**Reply by mivinci** [reply]
hmmm you know what, I think it looks good 👀 but I'm considering making the navbar turn into a dropdown menu automatically when it's on phones 🤔 it looks pretty bad right now if you have many links there 😥

**Reply by kranurag78** [reply]
I agree for this but one request from my side would be to have the basic things enabled by default like toggling dark theme and going to home. 
I think it will be great if we can have something like we allow 3 things to be there and after three things everything goes into the drop down. The first three can be decided as per the weight. 
Here's how it looks on the smartphone. 

<details>
<summary> screenshot </summary>

![image](https://user-images.githubusercontent.com/84301342/225668363-0a38b1f0-c31a-47ce-bb22-dafec05ee235.png)

</details>
